data_structures/src/chain.rs

```diff
@@ -299,6 +304,30 @@ pub enum Input {
     ValueTransfer(ValueTransferInput),
 }
 
+/// get OutputPointer from input
+impl Input {
+    pub fn get_output_pointer_from_input(&self) -> OutputPointer {
+        match self {
```

Comment:
This entire `match` looks redundant to me, let's add a TODO here for signaling this as a future refactor opportunity.